Libplanet.Action/ActionEvaluator.cs

```diff
@@ -233,9 +213,18 @@ public class ActionEvaluator : IActionEvaluator
             ITransaction? tx,
             IActionContext context,
             IAction action,
+            IStateStore stateStore,
             ILogger? logger = null)
         {
+            if (!context.PreviousState.Trie.Recorded)
+            {
+                throw new InvalidOperationException(
+                    $"Given {nameof(context)} must have its previous state's " +
+                    $"{nameof(ITrie)} recorded.");
+            }
```

Comment:
Is this check needed? we already check it on line `316...327`.

Reply:
It's a precaution for `EvaluateAction` to check both its input `ITrie` and output `ITrie`. I agree it seems a bit excessive. If we are to remove one, probably the latter one could be considered unnecessary.

Reply:
> latter one could be considered unnecessary.

I do not have a strong opinion about which one is more unnecessary. Feel free to choose. 😄

Reply:
https://github.com/planetarium/libplanet/pull/3520#discussion_r1414787260

Then we left both of them?